### ISSUE TITLE
"Empty Cards"

### ISSUE THREAD

**Issue by joliettetaci** [krassowski/Anki-Night-Mode#18]
After creating new cards when using this add-on, when you're NOT using this add-on to study those cards, those cards are sometimes problematic: **text with white color**. You might think that's an empty card but try highlighting those empty space, you'll see the text. >> This is a real problem when you are using **AnkiMobile (iOS)**, as the **cards appear empty**. It is pretty tedious to **edit the card one by one in order to fix the font colour issue** as I have only realized this issue after I have created an entire deck of 500+ cards. **Please help!** Also, thank you so much for such an amazing function - love the look of Anki so much better with this add-on!

**Comment by krassowski** [comment]
This is known issue and it is mainly present in the old version of Anki which uses long outdated and unsupported PyQt libraries. Do you use the new (still experimental) 2.1 Anki version or the old one from 2.0 branch?

A tip for quickly reverting formatting is to use `ctrl + r` shortcut.

You can also edit all cards at once easily. Do as follows:

0. Please export you deck into a file as a back-up!
1. Open browser (Browse or `b` shortcut)
2. Find any of affected cards and click on it; select the affected field
3. In editor click on "Edit HTML" button. In the new version it is under "hamburger" icon together with LaTex buttons, in old version it was arrow icon or something, one of the last buttons anyway.
4. A new window with HTML code will show up. You should find some text like:

```HTML
<font color="#ffffff">The text from your card</font>
```
or

```HTML
<font color="#fff">The text from your card</font>
```

or

```HTML
<font color="white">The text from your card</font>
```

or possibly even

```HTML
<span style="color:#ffffff">The text from your card</span>
```

5. Copy this text and save for later.
6. Let's go back to browser. Choose the deck you wish to edit and select all cards (`ctrl + a`)
7. In menu bar select "Edit -> Find and replace" (`ctr + alt + f` in new version, not sure about the old one)
8. Check "Treat the input as regular expression"
9. Type in `<font color="#ffffff">(.*)<\/font>` into "Find" field, using the template you found in 4/5 step. Pleas notice `\` before `/` and `(.*)` which replaced text from a card tested in step 2.
10. Type in `\1` into the "Replace with" field.
11. Press ok and verify if it worked.

Please let me know if it worked! I tested it with the new version where I intentionally created a few cards and colored texts so I could see if it works. I cannot install the old version on my system as it uses outdated and insecure libraries which is considered kind of a security thread. See https://bugs.launchpad.net/ubuntu/+source/python-qt4/+bug/1633819 which received "wont-fix" status:

>  Dmitry Shachnev (mitya57) wrote on 2016-12-27:
> 
> Won’t fix. The WebKit support was removed on purpose, because of security issues. It is highly recommended to migrate to PyQt5 (which does have WebKit support).

**Comment by krassowski** [comment]
Also you could try to achieve the same effect using [Clear Field Formatting (HTML) in Bulk](https://ankiweb.net/shared/info/1114708966) addon. I did not test it though. Anyway, please consider sharing you experience with other users as I see how annoying this bug can be. Thanks for you report!

**Comment by joliettetaci** [comment]
I'm using the old version (Anki 2.0). 

I've tried clicking reverting format (both using the button and the crt+r shortcut) previously (again, by editing one card at a time) - but the text still remains white/"transparent". 

It would only become black/"visible" if I changed the colour manually using "Set Foreground Colour (F7)". This is still a tedious work to do though..

I just tried the "Edit HTML" method, but only one line of words for each card that were initially transparent became visible.



Strangely though, now when I'm creating new cards (in the night mode) and reviewing it (in the normal mode), this problem doesn't exist! 

Not sure how the issue started in the first place.. Didn't notice the issue until I started using the mobile version, as even with the "transparent" text, there was no issue viewing the cards on desktop.

**Comment by krassowski** [comment]
> I just tried the "Edit HTML" method, but only one line of words for each card that were initially transparent became visible.

So I guess that to make it work on other lines too you will need to repeat steps 7-11 until all the lines are cleared. It still sound better than modifying 500+ cards manually. 

In the old version the problem was appearing most commonly when you were copying and pasting text between fields (and in several seemingly random moments too). I would recommend upgrading to 2.1 as soon as it will be officially released.

**Comment by joliettetaci** [comment]
![dark](https://user-images.githubusercontent.com/29819480/27767424-03d5c648-5f2a-11e7-9081-03271e566a12.JPG)

![normal](https://user-images.githubusercontent.com/29819480/27767423-03c24ab4-5f2a-11e7-81d9-529de73822fa.JPG)

![html editor](https://user-images.githubusercontent.com/29819480/27767441-82c819a6-5f2a-11e7-9aae-c3aa487fd744.JPG)

Ok, so the nightmare is back after editing new deck of cards that I just created, except this time I realized the bug is not as random and the `crt + r` method works while the "find and replace" doesn't seem so helpful as it is too complex (since this time round, 2 fields were affected instead of 1 like previously). 

And yes, it is definitely caused by copying and pasting text between fields (like you said)! I noticed the texts that I typed in remained black (see images above - "40" was typed in), while the texts I copied from other fields ("2014_0" and the entire choices field were copied from another field) in Anki became white; although copying from other sources seem ok (the question was copied from Word Doc).

> Also you could try to achieve the same effect using Clear Field Formatting (HTML) in Bulk addon.

So I tried the add-on that you suggested, and it works so well and is much easier to do then the "Find and Replace" method (yay!) - (although it still doesn't work for the old deck; I used the "Find and Replace" and repeated the steps till everything was cleared)

![clear formatting bulk](https://user-images.githubusercontent.com/29819480/27767468-2a110204-5f2b-11e7-852c-3f1df0493068.JPG) 
Just a note - I selected the option ("at all") initially but it messed up my text alignment, so I undo the action and selected the option ("Remain as New Line"). Great add-on.

For others facing the issue, try the add-on! Only if it doesn't work, try the "Find and Replace" method @krassowski suggested. For future note: try not to copy-and-paste from other fields as much as possible to avoid this problem. Cheers!

Thanks so much for the help @krassowski (and for the amazing add-on)!!! 👍
